src/GraphQL.ApiTests/BomTests.cs

```diff
@@ -0,0 +1,40 @@
+using System.Runtime.CompilerServices;
+
+namespace GraphQL.ApiTests;
+
+public class BomTests
+{
+    // https://github.com/graphql-dotnet/graphql-dotnet/pull/3477
+    [Fact]
+    public void Files_Should_Not_Use_BOM()
+    {
+        string GetPath([CallerFilePath] string path = "") => path; // <GIT_ROOT>\src\GraphQL.ApiTests\BomTests.cs
+
+        var gitRoot = new DirectoryInfo(GetPath()).Parent!.Parent!.Parent!;
```

Comment:
Then before performing any other checks, we should ensure that the sln exists in this folder.

Reply:
For what? There is no sln file in graphql root.

Reply:
We should just have some way of verifying we are reading the correct folder.

Reply:
Since the sln isn't in this folder, we should fix this so it isn't scanning the whole repo.  All the csproj and cs files are in the src folder right?  If that wasn't true, we should move the sln file.

Reply:
I intentionally want to scan all repo files. Why to lock onto sln location? The "correctness" of  the whole path is guaranteed by Roslyn of course.

Reply:
I don't think this PR should be merged.  It serves no valuable purpose.  It makes tests more complicated and performs unnecessary scanning of files.  As it is now, it's scanning folders that do not need to be scanned, such as the documentation files, and if someone changes the folder tree, may scan files outside of the repo.

As far as I'm aware, we generally have only had issues with csproj files, and rarely so.  (Notice that for the test to pass only some csproj files needed to be updated.)  I don't see why we are expending ANY time and effort on this.  Certainly this is not a KISS solution.  Better to work on actual features rather than making our own lives more difficult with tests that do not affect functionality, such as forcing us to fix bom errors when they arise.  We already have enough problems with tests we created, such as now needing to add ifdefs to `using` statements when they are not needed by all TFMs.  While formatting tests may be good, there is overkill, and this is one such example.

There are additional reasons not to merge this.  What if there are files in excluded directories within the scanned folder?  What about stuff like Nullability.Source which is not part of the project but yet will bring in cs files?  And how are we expected to change files if the bom is wrong -- open a hex editor??

If you really want to merge this, confine it to the src tree with a verification that GraphQL.sln exists there.  I don't think it should be merged at all and will just create extra work for us to maintain the project.

Reply:
>  It makes tests more complicated and performs unnecessary scanning of files.

😕  complicated? 14 lines of code and 0.3 sec for all repo scan. Nevertheless I can switch to scan src only, not a big deal.

Regarding the rest comments - I explained the reasoning before

>  These are parasitic changes that should not be. It's like if we constantly change spaces count from PR to PR.

Ideally dotnet-format tool should alert about BOM but it does not so I just filled a gap.

Reply:
Last days I was on the road so had no much time to work on it or other pending reviews. Will continue tomorrow.

Reply:
I thought one more time about your words about scanning and I think I have logical answer. Let's consider our editorconfig file:
```
[*]
insert_final_newline = true
charset = utf-8
```

1. It requires that ALL files should have utf-8 (no BOM) encoding without exceptions further down the file.
2. It requires this at the root repo level so it is intended to work even for docs, ci actions, whatever.

> confine it to the src tree

Then it would be logical to move editorconfig in `src` folder as well.

Reply:
All of my logic was explained in https://github.com/graphql-dotnet/graphql-dotnet/pull/3478#discussion_r1060301765 .  The editorconfig has no bearing on my reasoning.  I would not move the editorconfig.

I have no further comments.  As stated before, I think this test will simply add headache for the developers and should not be merged.